### PR TITLE
[Site-wide] Implement typeahead for global search bar (no production)

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR implements a __rough prototype__ of typeahead component into the search input of the global navigation. It uses a search.gov API, which is intended to be hit directly from the browser without any secret API key. In this PR, it is feature flagged to environments less than prod and the homepage only.

Per the [codepen](https://codepen.io/jenstrickland/pen/MWKmgoR) recommendation of our accessibility expert, this component uses a `datalist`. The CSS that applies to this element is very limited. I ended up not adding any because I couldn't find any CSS that worked.

This is for visibility purposes only. Once matured, this component will likely need to go through the collab cycle, and that will likely be delayed until we are satisfied with the data suggested by the search.gov API.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/10291

## Testing done
- Local testing of the new component behavior
- Added a unit test

## Screenshots
image

## Acceptance criteria
- [ ] Prototype of typeahead is created

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by rianfowler — APPROVED** [review]
this needs peer review approval before merging

File: src/platform/site-wide/user-nav/components/Typeahead.jsx
Comment (kelsonic) [inline src/platform/site-wide/user-nav/components/Typeahead.jsx:30]: Wow cool!

File: src/platform/site-wide/user-nav/components/Typeahead.jsx
Comment (kelsonic) [inline src/platform/site-wide/user-nav/components/Typeahead.jsx:9]: Could we add PropTypes for `userInput`?

File: src/platform/site-wide/user-nav/components/Typeahead.jsx
Comment (kelsonic) [inline src/platform/site-wide/user-nav/components/Typeahead.jsx:39]: Should we escape early with a `return null` if `this.props.userInput` is empty?

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉 Left a couple of comments but nothing major! Great work!!

File: src/platform/site-wide/user-nav/components/Typeahead.jsx
Comment (ncksllvn) [reply src/platform/site-wide/user-nav/components/Typeahead.jsx:40]: Hm I was thinking it would be okay for `<datalist>` to have no children, but you might be right...I'll check the docs

File: src/platform/site-wide/user-nav/components/Typeahead.jsx
Comment (ncksllvn) [reply src/platform/site-wide/user-nav/components/Typeahead.jsx:10]: Good idea, done!

File: src/platform/site-wide/user-nav/components/Typeahead.jsx
Comment (ncksllvn) [reply src/platform/site-wide/user-nav/components/Typeahead.jsx:40]: I think this is okay to be empty because the [docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist) say `zero or more <option> elements.`

**Pull request merged**
